lib/recurly/pager.php

```diff
@@ -9,25 +9,28 @@
 abstract class Recurly_Pager extends Recurly_Base implements Iterator, Countable
 {
   private $_position = 0;    // position within the current page
-  protected $_count = null;  // total number of records
   protected $_objects;       // current page of records
 
   /**
-   * Number of records in this list.
+   * Number of records in this list. This calls
+   * the server with a HEAD request every time the function
+   * is called.
+   *
+   * @param Recurly_Client Optional client useful for mocking
    * @return integer number of records in list
    */
-  public function count() {
-    if (!isset($this->_count)) {
-      if (isset($this->_objects)) {
-        $this->_count = count($this->_objects);
-      } elseif (isset($this->_href)) {
-        // Don't bother with the HEAD request the server takes the same amount
-        // of time to generate them so, might as well just get the results at
-        // the same time.
-        $this->_loadFrom($this->_href);
-      }
+  public function count($client = null) {
```

Comment:
I don't think the count method should take a client. That should be included in the constructor when the pager is instantiated. We should only be injecting it into static methods.